src/mlpack/tests/saga_test.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file saga_test.cpp
+ * @author Prabhat Sharma
+ *
+ * Test file for SAGA.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/saga/saga.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+#include "test_function_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::optimization;
+
+BOOST_AUTO_TEST_SUITE(SAGATest);
```

Comment:
Do you think you could add two more tests, you could pick from: https://github.com/mlpack/mlpack/tree/master/src/mlpack/core/optimizers/problems.

Reply:
@zoq  I've added Rosenbrock test for both SAGA and SVRG. I think It'll be sufficient? One thing I've noticed in the process is that SAGA requires much smaller step size as compared to SVRG even if both are quite similar. Another thing is that these both cannot pass `sgd_test_function` I don't know why but I've tried experimenting with many values but both couldn't pass till the end